csharp/src/Ice/OutgoingRequestFrame.cs

```diff
@@ -302,12 +305,39 @@ internal override void Finish()
                 }
             }
 
+            var ostr = new OutputStream(proxy.Protocol.GetEncoding(), Data);
+
             if (Protocol == Protocol.Ice1)
             {
+                // Marshaled "by hand" to avoid allocating a string[] for the facet and a new dictionary for the
+                // context.
+                Identity.IceWrite(ostr);
+                ostr.WriteFacet11(Facet);
+                ostr.WriteString(Operation);
+                ostr.Write(IsIdempotent ? OperationMode.Idempotent : OperationMode.Normal);
                 ostr.WriteDictionary(_initialContext,
                                      OutputStream.IceWriterFromString,
                                      OutputStream.IceWriterFromString);
             }
+            else
+            {
+                Debug.Assert(Protocol == Protocol.Ice2);
+
+                // Marshaled "by hand" to avoid allocating a string[] for the location.
+                BitSequence bitSequence = ostr.WriteBitSequence(3);
+                Identity.IceWrite(ostr);
+                if (bitSequence[0] = (Facet.Length > 0))
```

Comment:
Is this correct? I find it more clear how we do it in the generated code

```
if (Facet.Length > 0)
{
    ostr.WriteString(Facet);
}
else
{
    bitSequence[0] = false;
}
```

Reply:
It's correct, but I fixed it to use the same style as the generated code.